lib/connections/quic_listen.go

```diff
@@ -59,7 +59,8 @@ func (t *quicListener) OnNATTypeChanged(natType stun.NATType) {
 func (t *quicListener) OnExternalAddressChanged(address *stun.Host, via string) {
 	var uri *url.URL
 	if address != nil {
-		uri = &(*t.uri)
+		uri = &url.URL{}
+		*uri = *t.uri
```

Comment:
Why does `uri` need to be a pointer? There's nil checks, but wouldn't the line below (`uri.Host`) anyway blow up if `t.uri` was nil?

Reply:
.Host would not blow up as `&(*t.uri)` is never null, so is `uri = &uri.URL{}`

There is a distinction between not having resolved an address yet (nil), and having resolved the addres (address not nil, hence current uri is t.uri + new host that is resolved).

Reply:
Yeah right. Personally I'd consider it slightly simpler if `uri` wasn't a pointer. The condition later at line 72 would then just need to check `address == nil` instead. Anyways definitely nothing important.

Generally lgtm, maybe lets wait for Jakob to have a quick look as he brought this up.